Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabSystemComponent.cpp

```diff
@@ -92,25 +92,28 @@ namespace AzToolsFramework
 
         AZStd::unique_ptr<Instance> PrefabSystemComponent::CreatePrefab(
             const AZStd::vector<AZ::Entity*>& entities, AZStd::vector<AZStd::unique_ptr<Instance>>&& instancesToConsume,
-            AZ::IO::PathView filePath, AZStd::unique_ptr<AZ::Entity> containerEntity)
+            AZ::IO::PathView filePath, AZStd::unique_ptr<AZ::Entity> containerEntity, InstanceOptionalReference parent,
+            bool shouldCreateLinks)
         {
-            AZStd::unique_ptr<Instance> newInstance = AZStd::make_unique<Instance>(AZStd::move(containerEntity));
-            return CreatePrefab(entities, AZStd::move(instancesToConsume), filePath, AZStd::move(newInstance), true);
-        }
 
-        AZStd::unique_ptr<Instance> PrefabSystemComponent::CreatePrefabUnderParent(
-            const AZStd::vector<AZ::Entity*>& entities, AZStd::vector<AZStd::unique_ptr<Instance>>&& instancesToConsume,
-            AZ::IO::PathView filePath, InstanceOptionalReference parent)
-        {
-            AZ_Assert(parent.has_value(), "Prefab - Null parent instance passed in during CreatePrefabUnderParent");
+            AZStd::unique_ptr<Instance> newInstance;
+
+            if (parent.has_value())
+            {
+                newInstance = AZStd::make_unique<Instance>(parent);
+            }
+            else
+            {
+                newInstance = AZStd::make_unique<Instance>(AZStd::move(containerEntity));
```

Comment:
why are we only passing the container entity to the instance if parent is not provided? I don't understand the correlation between the two. A user may choose to provide both a parent instance and a container entity. I'd expect that logic to be if(parent)->setParent() and if (containerEntity), setContainerEntity().

Reply:
Refactored instance constructors.